src/datahike/query.cljc

```diff
@@ -204,7 +204,15 @@
 
 (defmulti -lesser?
   {:arglists '([value & more])}
-  (fn [value & more] (class value)))
+  (fn [value & _more] (class value)))
+
+(defmethod -lesser? java.lang.String [^String s0 & more]
+  (reduce (fn [res [s1 ^String s2]]
+            (if (neg? (compare s1 s2))
+              res
+              (reduced false)))
+          true
```

Comment:
Is it sound to have both `-lesser?` and `-greater?` default to `true`?

Reply:
It's the same with `<` and `>`, so I would say: yes

Reply:
I think this is mathematically questionable, because `<` and `>` are strict binary relations which cannot both be true for the same arguments, but if this is expected behaviour in Clojure then this is what it is. It would be better to always require at least 2 arguments to assign meaning to `<` or `>`. This is different for `+` or `*` defaulting to the neutral elements in their group for no arguments, which I think is mathematically reasonable.